### PR TITLE
docs: add some setup instructions

### PR DESCRIPTION
**Pull request by jquense**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -40,6 +40,45 @@ api.add_resource("/widgets", WidgetListView, WidgetView)
...
+```sh
+pyenv virtualenv 3.8.3 flask-resty
+pyenv local flask-resty
+```
+
+Install dependencies:
+
+```sh
+# optionally include [jwt] for the full test suite, but requires cryptography,
+# and it's dependent binary libraries
```
Comment (sloria) [inline README.md:56]:
```suggestion
#  and its dependent binary libraries
```

File: README.md [hunk 2]
```diff
@@ -40,6 +40,45 @@ api.add_resource("/widgets", WidgetListView, WidgetView)
...
+```sh
+sphinx-build docs/ docs/_build
+```
+
+For auto rebuilds and watching, `sphinx-autobuild` can be used:
+
+```sh
+pip install sphinx-autobuild
+sphinx-autobuild --open-browser docs/ docs/_build -z flask_resty -s 2
+```
```
Comment (sloria) [inline README.md:80]: suggestion: Probably better to just document one way to do things. Most people will want autobuild. Also, you can use tox for this:

```
tox -e watch-docs
```

which means you don't need to install anything extra yourself.

Here's how it's documented in marshmallow: https://marshmallow.readthedocs.io/en/stable/contributing.html#documentation

File: README.md [hunk 3]
```diff
@@ -40,6 +40,45 @@ api.add_resource("/widgets", WidgetListView, WidgetView)
...
+# optionally include [jwt] for the full test suite, but requires cryptography,
+# and it's dependent binary libraries
+pip install -e .[testing,docs]
+```
+
+### Running tests
+
+Tests can be run for your current enviroment by running pytest directly: `pytest`.
+
+Or run `tox` to run the fulle suite of lint checks and tests against flask-resty's supported
```
Comment (sloria) [inline README.md:64]:
```suggestion
Or run `tox` to run the full suite of lint checks and tests against flask-resty's supported
```

File: README.md [hunk 4]
```diff
@@ -40,6 +40,45 @@ api.add_resource("/widgets", WidgetListView, WidgetView)
...
+environments.
+
```
Comment (sloria) [inline README.md:66]: suggestion: Might be useful to document `tox -e lint` here as well.

FWIW, here's how marshmallow documents it: https://marshmallow.readthedocs.io/en/stable/contributing.html#running-tests

File: README.md [hunk 5]
```diff
@@ -40,6 +40,45 @@ api.add_resource("/widgets", WidgetListView, WidgetView)
 
 Documentation is available at https://flask-resty.readthedocs.io/.
 
+## Running locally
+
+Ideally, create a virtual env for local development, using your tool of choice, e.g.:
+
+```sh
+pyenv virtualenv 3.8.3 flask-resty
```
Comment (sloria) [inline README.md:48]: suggestion: Might as well use the latest version

```suggestion
pyenv virtualenv 3.8.5 flask-resty
```

File: README.md [hunk 6]
```diff
@@ -40,6 +40,45 @@ api.add_resource("/widgets", WidgetListView, WidgetView)
 
 Documentation is available at https://flask-resty.readthedocs.io/.
 
+## Running locally
+
+Ideally, create a virtual env for local development, using your tool of choice, e.g.:
```
Comment (sloria) [inline README.md:45]:
```suggestion
Create a virtualenv for local development using your tool of choice. If you're using pyenv and pyenv-virtualenv:
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
+# Contributing
+
+Contributions are actively welcome!
+
+## Local Setup
+
+Ideally, create a virtual env for local development, using your tool of choice, e.g.:
```
Comment (jquense) [inline CONTRIBUTING.md:7]:
```suggestion
Create a virtualenv for local development using your tool of choice. If you're using pyenv and pyenv-virtualenv:
```

**Comment by sloria** [comment]
> can we include this in the auto-generated docs as with https://github.com/marshmallow-code/marshmallow/blob/dev/docs/contributing.rst ?

Not if you want to use Markdown...it's not a bad idea to do eventually, but I think it's fine to just have CONTRIBUTING.md. I do that for [some](https://github.com/sloria/environs/blob/master/CONTRIBUTING.md) of my projects when I'm to lazy to relearn RST.

**Comment by jquense** [comment]
gonna stick this in for now, since i have no idea how to write RST or figure that part out :P

**Pull request merged**